packages/hardhat-chai-matchers/test/helpers.ts

```diff
@@ -153,3 +156,20 @@ export async function runFailedAsserts({
     failedAssert(matchers[method].staticCall(...args))
   ).to.be.rejectedWith(AssertionError, failedAssertReason);
 }
+
+/**
+ * Used to test that non-chainable matchers like changeEtherBalance,
+ * changeTokenBalance and reverted throw an error when used along with other
+ * non-chainable matchers.
+ */
+function supportNonChainableMatcher(
```

Comment:
I'm not sure about this. I'd rather use realistic tests, because this feels too white-boxy to me.

Reply:
I think the transparency is needed here. Chaining it to other matchers that could or could not be incompatible is less clear. Instead, this expresses clearly the test intent.